pyvista/core/utilities/points.py

```diff
@@ -459,3 +464,161 @@ def vector_poly_data(orig, vec):
+    in the off-diagonal. This indicates that the first principal axis is
+    aligned with the x-axis, the second with the y-axis, and third with
+    the z-axis. This is expected, since the mesh is already axis-aligned.
+
+    However, since the signs of the principal axes are arbitrary, the
+    first and third axes in this case have a negative direction.
+
+    Show the standard deviation along each axis.
+
+    >>> std
+    array([3.014956 , 1.507478 , 0.7035637], dtype=float32)
+
+    Compare this to using :meth:`numpy.std` for the computation.
+
+    >>> np.std(mesh.points, axis=0)
+    pyvista_ndarray([3.0149572, 1.5074761, 0.7035699], dtype=float32)
+
+    Convert the values to proportions for analysis.
+
+    >>> std / sum(std)
+    array([0.5769149 , 0.28845742, 0.1346276 ], dtype=float32)
+
+    From this result, we can determine that the axes explain approximately
+    58%, 29%, and 13% of the total variance in the points, respectively.
+
+    Let's compare this to the proportions of the known radii of the ellipsoid.
+
+    >>> radii / sum(radii)
+    array([0.6, 0.3, 0.1])
+
+    Note how the two ratios are similar, but do not match exactly. This is
+    because the points of the ellipsoid are prolate and are denser near the
+    poles. If the points were normally distributed, however, the proportions
+    would match exactly.
+
+    Create an array of normally distributed points scaled along the x-y-z axes.
+    Use the same scaling as the radii of the ellipsoid from the previous example.
+
+    >>> normal_points = rng.normal(size=(1000, 3))
+    >>> scaled_points = normal_points * radii
+    >>> axes, std = pv.principal_axes(scaled_points, return_std=True)
+    >>> axes
+    array([[-0.99997578,  0.00682346,  0.00136972],
+           [ 0.00681368,  0.99995213, -0.00702282],
+           [-0.00141757, -0.00701331, -0.9999744 ]])
+
+    Once again, the axes have ones along the diagonal as expected since the
+    points are already axis-aligned. Now let's examine the standard deviation
+    and compare the relative proportions.
+
+    >>> std
+    array([5.94466738, 2.89590334, 1.02103169])
+
+    >>> std / sum(std)
+    array([0.60280948, 0.29365444, 0.10353608])
+
+    >>> radii / sum(radii)
+    array([0.6, 0.3, 0.1])
+
+    Since the points are normally distributed, the relative proportion of
+    the standard deviation matches the scaling of the axes almost perfectly.
+    """
+    points = _validation.validate_arrayNx3(points)
+
+    points_centered = points - np.mean(points, axis=0)
+    eig_vals, eig_vectors = np.linalg.eigh(points_centered.T @ points_centered)
+    axes = eig_vectors.T[::-1]  # columns, ascending order -> rows, descending order
+
+    # Ensure axes form a right-handed coordinate frame
+    if np.linalg.det(axes) < 0:
+        axes[2] *= -1
+
+    if return_std:
+        # Compute standard deviation and swap order from ascending -> descending
+        std = np.sqrt(eig_vals / len(points), out=eig_vals)[::-1]
```

Comment:
This can be reverted, left over from investigating the array type of the eigenvalues.
```suggestion
        std = np.sqrt(eig_vals / len(points))[::-1]
```